File: leetcode/easy/two-sum/go/two_sum.go
package leetcode

func twoSum(nums []int, target int) []int {
	indices := make(map[int]int)
	var result []int
	for i, num := range nums {
		if j, ok := indices[target-num]; ok {
			result = append(result, j, i)
			break
		}
		indices[num] = i
	}
	return result
}

File: leetcode/easy/two-sum/haskell/twoSum.hs
-- haskell ☺️

twoSum :: [Int] -> Int -> (Int, Int)
twoSum nums target = [(x, y) | x <- i, y <- i, x < y && nums !! x + nums !! y == target] !! 0
    where i = [0..length nums - 1]
